samples/fsharp/getting-started/Regression_BikeSharingDemand/BikeSharingDemand/BikeSharingDemandConsoleApp/ModelScoringTester.fs

```diff
@@ -6,37 +6,38 @@ open DataStructures
 open Microsoft.ML
 open System.Globalization
 
+let parseSingle (s:string) = Single.Parse(s, CultureInfo.InvariantCulture)
+
 let readSampleDataFromCsvFile dataLocation numberOfRecordsToRead =
-    File.ReadLines(dataLocation)
+    File.ReadLines dataLocation
     |> Seq.skip 1
-    |> Seq.filter(fun x -> not(String.IsNullOrEmpty(x)))
-    |> Seq.map(fun x -> x.Split(','))
-    |> Seq.map(fun x -> {
-        Season = Single.Parse(x.[2], CultureInfo.InvariantCulture)
-        Year = Single.Parse(x.[3], CultureInfo.InvariantCulture)
-        Month = Single.Parse(x.[4], CultureInfo.InvariantCulture)
-        Hour = Single.Parse(x.[5], CultureInfo.InvariantCulture)
-        Holiday = Single.Parse(x.[6], CultureInfo.InvariantCulture)
-        Weekday = Single.Parse(x.[7], CultureInfo.InvariantCulture)
-        WorkingDay = Single.Parse(x.[8], CultureInfo.InvariantCulture)
-        Weather = Single.Parse(x.[9], CultureInfo.InvariantCulture)
-        Temperature = Single.Parse(x.[10], CultureInfo.InvariantCulture)
-        NormalizedTemperature = Single.Parse(x.[11], CultureInfo.InvariantCulture)
-        Humidity = Single.Parse(x.[12], CultureInfo.InvariantCulture)
-        Windspeed = Single.Parse(x.[13], CultureInfo.InvariantCulture)
-        Count = Single.Parse(x.[16], CultureInfo.InvariantCulture)
-    })
+    |> Seq.filter (fun x -> not (String.IsNullOrEmpty x))
+    |> Seq.map (fun x ->
+        let fields = x.Split(',')
+        { Season = parseSingle fields.[2]
```

Comment:
maybe just drop the `parseSingle` def and use `float32`, its basically equivalent.

Reply:
I don't think that's using InvariantCulture.

Reply:
It's using InvariantCulture, the difference is `NumberStyles.AllowThousands` is not set in favor of striping all underscores prior to parsing https://github.com/dotnet/fsharp/blob/f9ea5824a321a0df73675ad392458af79297f092/src/fsharp/FSharp.Core/prim-types.fs#L3578